src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -162,5 +162,7 @@ sealed class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering:
       iterator.foreach { case (key, value) => update(key, f(key, value)) }
       this
     }
+
+    override def clone() = super.clone().rangeImpl(from, until)
```

Comment:
Only the elements returned by this instance's `iterator` are present in the cloned instance, not the whole underlying collection - `super.clone()` is defined as `empty ++= repr.seq`.

The example in the comment above - in particular the `view.clone()` line - shows the behavior of the old implementation (`super.clone()`) and you can see that it returns only the elements of the view. The only thing added here is the `rangeImpl` call, which when called does not filter any element, but will filter future set insertions.